src/khoj/routers/api.py

```diff
@@ -444,33 +438,56 @@ def chat_init(
     return {"status": "ok", "response": meta_log.get("chat", [])}
 
 
-@api.get("/chat", response_class=StreamingResponse)
+@api.get("/chat", response_class=Response)
 async def chat(
     request: Request,
-    q: Optional[str] = None,
+    q: str,
     n: Optional[int] = 5,
     client: Optional[str] = None,
+    stream: Optional[bool] = False,
     user_agent: Optional[str] = Header(None),
     referer: Optional[str] = Header(None),
     host: Optional[str] = Header(None),
-) -> StreamingResponse:
-    def _save_to_conversation_log(
-        q: str,
-        gpt_response: str,
-        user_message_time: str,
-        compiled_references: List[str],
-        inferred_queries: List[str],
-        meta_log,
-    ):
-        state.processor_config.conversation.chat_session += reciprocal_conversation_to_chatml([q, gpt_response])
-        state.processor_config.conversation.meta_log["chat"] = message_to_log(
-            q,
-            gpt_response,
-            user_message_metadata={"created": user_message_time},
-            khoj_message_metadata={"context": compiled_references, "intent": {"inferred-queries": inferred_queries}},
-            conversation_log=meta_log.get("chat", []),
-        )
+) -> Response:
+    _perform_chat_checks()
+    compiled_references, inferred_queries = await _get_top_k_results(request, q, (n or 5))
+
+    gpt_response = _generate_chat_response(
+        request,
+        q,
+        client,
+        user_agent,
+        referer,
+        host,
+        meta_log=state.processor_config.conversation.meta_log,
+        compiled_references=compiled_references,
+        inferred_queries=inferred_queries,
+    )
+
+    if gpt_response is None:
+        return Response(content="No response from GPT", media_type="text/plain", status_code=500)
+
+    if stream:
+        return StreamingResponse(gpt_response, media_type="text/event-stream", status_code=200)
+
+    if isinstance(gpt_response, str):
+        return Response(content=gpt_response, media_type="text/plain", status_code=200)
```

Comment:
Shouldn't this result in status_code=500 as `gpt_response` is string only on error? If so, can this be merged with the above `gpt_response is None` conditional?

Reply:
Very good point.